packages/twenty-server/src/engine/metadata-modules/field-metadata/services/field-metadata-related-records.service.ts

```diff
@@ -19,24 +27,37 @@ type Differences<T> = {
   deleted: T[];
 };
 
+type GetOptionsDifferences = Differences<
+  FieldMetadataDefaultOption | FieldMetadataComplexOption
+>;
+
+export type SelectFieldMetadataEntity = FieldMetadataEntity<
+  | FieldMetadataType.SELECT
+  | FieldMetadataType.RATING
+  | FieldMetadataType.MULTI_SELECT
+>;
+
 @Injectable()
 export class FieldMetadataRelatedRecordsService {
   constructor(
     private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
   ) {}
 
   public async updateRelatedViewGroups(
-    oldFieldMetadata: FieldMetadataEntity,
-    newFieldMetadata: FieldMetadataEntity,
+    oldFieldMetadata: SelectFieldMetadataEntity,
+    newFieldMetadata: SelectFieldMetadataEntity,
   ): Promise<void> {
+    // TODO legacy should support multi-select and rating ?
     if (
       !isSelectFieldMetadataType(newFieldMetadata.type) ||
       !isSelectFieldMetadataType(oldFieldMetadata.type)
     ) {
       return;
     }
-
-    const views = await this.getFieldMetadataViews(newFieldMetadata);
+    const views = await this.getFieldMetadataViewEntity(
```

Comment:
getFieldMetadataViewEntities would be clearer, wouldn't it ?

Reply:
The function only returns one entity

Reply:
In my mind the `entity` is either `viewFilter` or `viewGroup` and It the methods returns the occurrences of this entity 🤔

Reply:
I kept the previously named `views` const but indeed it's not accurate, could be renamed to `viewGroups` or`groups`